index.d.ts

```diff
@@ -0,0 +1,313 @@
+             * (detected only after sending a failed notification to the device).
+             */
+            unsubscribed?: boolean;
+        };
+
+        /**
+         * Pending notifications that have not yet been delivered to the device.
+         *
+         * Only supported for Android & Electron devices.
+         * iOS & Web Push make it impossible to track notification delivery and will always return an empty array.
+         * */
+        pending_notifications: Array<{
+            /** The push notification's unique ID. */
+            id: string;
+
+            /** The creation date of the push notification (unix timestamp). */
+            date: number;
+
+            /** The push notification payload data. */
+            payload: unknown;
+
+            /** The push notification expiration date (unix timestamp). */
+            expiration: number;
+        }>;
+
+        /** An array of topics the device is subscribed to. */
+        subscriptions: Array<string>;
+    }
+
+    interface DevicePresenceInfo {
+        /** The device token linked to this presence object. */
+        id: string;
+
+        /** The device's current connectivity status. */
+        online: boolean;
+
+        /** The device's last connection date (unix timestamp). */
+        last_active: number;
+
+        /**
+         * When returned, the iOS user has uninstalled your app
+         * (detected only after sending a failed notification to the device).
+         */
+        uninstalled?: boolean;
+
+        /**
+         * When returned, the Web Push user has unsubscribed from notifications from your website
+         * (detected only after sending a failed notification to the device).
+         */
+        unsubscribed?: boolean;
+    }
+
+    interface TopicStatus {
+        /** The Pub/Sub topic name. */
+        name: string;
+
+        /** The Pub/Sub topic subscriber count. */
+        subscribers: number;
+    }
+
+    interface TopicSuscribers {
+        /** Array of device tokens currently subscribed to the provided Pub/Sub topic, limited to 1,000 per page. */
+        subscribers: Array<string>;
+
+        /** A link to fetch the next page of results, in case there are more than 1,000 subscribers for this topic. */
+        pagination: string;
+    }
+
+    export default class Pushy {
+        constructor(apiKey: string);
+
+        /**
+         * Send push notification
+         * @see {@link https://pushy.me/docs/api/send-notifications}
+         *
+         * @param data JSON notification data object
+         * @param recipient one or array of device token OR one or array of topics
+         * @param options Extra options for the notification
+         *
+         * @returns pushId
+         */
+        sendPushNotification(
+            data: unknown,
+            recipient: string | Array<string>,
+            options?: SendPushNotificationOptions
+        ): Promise<string>;
```

Comment:
Should we document the optional `callback` parameter? 
```suggestion
        sendPushNotification(
            data: unknown,
            recipient: string | Array<string>,
            options?: SendPushNotificationOptions,
            callback?: (error: Error | null, pushId: string) => void
        ): Promise<string>;
```

Reply:
Yes, we think this would be preferable.